platform/editor.mimelookup/src/org/netbeans/api/editor/mimelookup/MimePath.java

```diff
@@ -278,6 +278,12 @@ public static boolean validate(CharSequence type, CharSequence subtype) {
      * @since 1.7
      */
     public static boolean validate(CharSequence path) {
+        String pathString = path == null ? null : path.toString();
```

Comment:
Doesn't `parseImpl()` immediately blow up with `null` anyway? Could simplify and let it throw NPE here?